pv_site_api/app.py

```diff
@@ -0,0 +1,439 @@
+        """This route gets the status of the system.
+
+        It's mostly used by OCF to
+        make sure things are running smoothly.
+        """
+
+        if is_fake:
+            return make_fake_status()
+
+        status = get_latest_status(session=session)
+
+        status = PVSiteAPIStatus(status=status.status, message=status.message)
+
+        return status
+
+    @app.get("/")
+    def get_api_information():
+        """
+        ####  This route returns basic information about the Nowcasting PV Site API.
+
+        """
+
+        logger.info("Route / has been called")
+
+        return {
+            "title": title,
+            "version": pv_site_api.__version__,
+            "progress": "The Nowcasting PV Site API is still underconstruction.",
+        }
+
+    # @app.get("/favicon.ico", include_in_schema=False)
+    # def get_favicon() -> FileResponse:
+    #     """Get favicon"""
+    #     return FileResponse(f"/favicon.ico")
+
+    @app.get("/nowcasting.png", include_in_schema=False)
+    def get_nowcasting_logo() -> FileResponse:
+        """Get logo"""
+        folder = os.path.dirname(os.path.abspath(__file__))
+        return FileResponse(f"{folder}/nowcasting.png")
+
+    @app.get("/docs", include_in_schema=False)
+    def redoc_html():
+        """### Render ReDoc with custom theme options included"""
+        return get_redoc_html_with_theme(
+            title=title,
+        )
+
+    # OpenAPI (ReDoc) custom theme
+    def custom_openapi():
+        """Make custom redoc theme"""
+        if app.openapi_schema:
+            return app.openapi_schema
+
+        openapi_schema = get_openapi(
+            title=title,
+            version=pv_site_api.__version__,
+            description=description,
+            contact={
+                "name": "Nowcasting by Open Climate Fix",
+                "url": "https://nowcasting.io",
+                "email": "info@openclimatefix.org",
+            },
+            license_info={
+                "name": "MIT License",
+                "url": "https://github.com/openclimatefix/nowcasting_api/blob/main/LICENSE",
+            },
+            routes=app.routes,
+        )
+        openapi_schema["info"]["x-logo"] = {"url": "/nowcasting.png"}
+        app.openapi_schema = openapi_schema
+        return app.openapi_schema
+
+    app.openapi = custom_openapi
+
+
+def create_app(*, config: Config, get_session: Callable, auth: Callable) -> FastAPI:
+    app = FastAPI(docs_url="/swagger", redoc_url=None)
+
+    sentry_sdk.init(
+        dsn=config.SENTRY_DSN,
+        environment=config.ENVIRONMENT,
+        traces_sampler=0.0 if config.ENVIRONMENT == "local" else traces_sampler,
+    )
+
+    _add_cors(app, origins=config.CORS_ORIGINS)
+
+    _add_base_routes(app, is_fake=config.FAKE, get_session=get_session)
+
+    site_router = _create_site_router(is_fake=config.FAKE, get_session=get_session, auth=auth)
+    app.include_router(site_router)
```

Comment:
I can understand why you did this, 

I wonder if we should make sure the code is simple, and then have slightly more complex test setup, than, making the code more complicated. I think in general we should try and stick with the FastAPI way of doing things, as it will be easier to maintain, develop, e.t.c